keep/common/management/commands/update_pidman_title.py

```diff
@@ -0,0 +1,247 @@
+           pidman (DjangoPidmanRestClient): PidmanRestClient that can be used
+            to interact with Pidman
+        """
+        # update progress on the screen
+        sys.stdout.write("Starting %s tasks \n" % task_name)
+        sys.stdout.flush()
+
+        # initialize counters
+        total_count = len(objects)
+        fail_count = 0
+        change_count = 0
+        nochange_count = 0
+
+        # create log files
+        change_log = open(("%s/%s" % (self.output_path, "change.csv")), "w+")
+        nochange_log = open(("%s/%s" % (self.output_path, "nochange.csv")), "w+")
+        summary_log = open(("%s/%s" % (self.output_path, "summary.log")), "w+")
+
+        # iterate through all items in collection
+        for index, item in enumerate(objects):
+            try:
+                pidman_label = pidman.get_ark(item.noid)['name']
+                # TODO: to be safe the actual code to update is not included here
+                # will verify the environments before we accidentally start
+                # a detrimental process
+                # if self.is_dry_run:
+                #     #TODO skip the actual update codes
+                # else:
+                #     #TODO actually update (BE CAREFUL)
+
+                # when the names are not the same
+                if (pidman_label != item.label):
+                    change_count += 1
+                    # change_log.write("[TIME]: %s, [PID]: %s, [FEDORA_LABEL]: %s, [PIDMAN_LABEL]: %s \n" % (time.strftime("%Y%m%d %H:%M:%S", time.localtime()), item.pid, item.label, pidman_label) )
+                    change_log.write("%s, %s, %s, %s, %s\n" % \
+                        (time.strftime("%Y-%m-%d %H:%M:%S", \
+                        time.localtime()), \
+                        task_name, \
+                        item.pid, \
+                        item.label, \
+                        pidman_label))
+                # when the names are the same
+                else:
+                    nochange_count += 1
+                    # nochange_log.write("[TIME]: %s, [PID]: %s, [FEDORA_LABEL]: %s, [PIDMAN_LABEL]: %s \n" % (time.strftime("%Y%m%d %H:%M:%S", time.localtime()), item.pid, item.label, pidman_label) )
+                    nochange_log.write("%s, %s, %s, %s, %s\n" % \
+                        (time.strftime("%Y-%m-%d %H:%M:%S", \
+                        time.localtime()), \
+                        task_name, \
+                        item.pid, \
+                        item.label, \
+                        pidman_label))
+
+                # #TODO: remove this: write in the success log
+                # success_log.write("[TIME]: %s, [PID]: %s, [FEDORA_LABEL]: %s, [PIDMAN_LABEL]: %s \n" % (time.strftime("%Y%m%d %H:%M:%S", time.localtime()), item.pid, item.label, pidman_label) )
+                # success_count += 1
+
+                # update progress on screen
+                msg = "Object %i of %i for %s" % (index, len(objects)-1, task_name)
```

Comment:
This is a clever way to display progress, but I highly recommend the progressbar/progressbar2 python library, which we're using elsewhere.  Here's an example [manage script in readux](https://github.com/emory-libraries/readux/blob/d8f7fe8d0ac9a1dcccc300badb7ff4418baaab34/readux/books/management/commands/update_page_arks.py) with a progressbar; there are other examples in readux, eulfedora, and probably other places as well.